specs/test_formats/ssz_generic/README.md

```diff
@@ -1,20 +1,197 @@
 # SSZ, generic tests
 
 This set of test-suites provides general testing for SSZ:
- to instantiate any container/list/vector/other type from binary data.
+ to decode any container/list/vector/other type from binary data, encode it back, and compute the hash-tree-root.
 
-Since SSZ is in a development-phase, the full suite of features is not covered yet.
-Note that these tests are based on the older SSZ package.
-The tests are still relevant, but limited in scope:
- more complex object encodings have changed since the original SSZ testing.
+This test collection for general-purpose SSZ is experimental.
+The `ssz_static` suite is the required minimal support for SSZ, and should be prioritized.
 
-A minimal but useful series of tests covering `uint` encoding and decoding is provided.
-This is a direct port of the older SSZ `uint` tests (minus outdated test cases).
+The `ssz_generic` tests are split up into different handler, each specialized into a SSZ type:
 
-Test format documentation can be found here: [uint test format](./uint.md).
+- Vectors
+    - `basic_vector`
+    - `complex_vector` *not supported yet*
+- List
+    - `basic_list` *not supported yet*
+    - `complex_list` *not supported yet*
+- Bitfields
+    - `bitvector`
+    - `bitlist`
+- Basic types
+    - `boolean`
+    - `uints`
+- Containers
+    - `containers`
 
-*Note*: The current Phase 0 spec does not use larger uints, and uses byte vectors (fixed length) instead to represent roots etc.
-The exact uint lengths to support may be redefined in the future.
 
-Extension of the SSZ tests collection is planned, with an update to the new spec-maintained `minimal_ssz.py`;
- see CI/testing issues for progress tracking.
+## Format
+
+For each type, a `valid` and a `invalid` suite is implemented.
+The cases have the same format, but those in the `invalid` suite only declare a subset of the data a test in the `valid` declares.
+
+Each of the handlers encodes the SSZ type declaration in the file-name. See [Type Declarations](#type-declarations).
+
+### `valid`
+
+Valid has 3 parts: `meta.yaml`, `serialized.ssz`, `value.yaml`
+
+### `meta.yaml`
+
+Valid ssz objects can have a hash-tree-root, and for some types also a signing-root.
+The expected roots are encoded into the metadata yaml:
+
+```yaml
```

Comment:
It seems we've avoided putting "inputs" and "results" in the `meta.yaml` file for most other suites. 
The genesis tests for example have the eth1_timestamp (an input) in it's own file instead of just putting in meta. Thoughts on including "inputs" and/or "results" in the meta file?

Reply:
Not too sure here. It's also an identifier for the actual contents. I may just put it into a `roots.yaml` instead.